src/main/active_connection.rs

```diff
@@ -412,7 +412,9 @@ mod tests {
                 let state = unwrap!(core.get_state(sock_token));
                 let mut state = state.borrow_mut();
                 let active_conn = unwrap!(state.as_any().downcast_mut::<ActiveConnection>());
-                wait_for(sock_token, &poll);
```

Comment:
I don't see why this should fire at all in a deterministic way. You have not registered this socket (`peer1_sock`) with poll to hear about readable readiness so even though the listener writes something, there won't be an notification fired.